### PR TITLE
Completely replace SMS sending code.

### PR DESCRIPTION
**Pull request by meshy**
...and tidy up some.

### CODE REVIEW

**Comment by meshy** [comment]
@incuna/backend better?

**Comment by meshy** [comment]
Any objections to making this public and adding it to pypi?

**Comment by maxpeterson** [comment]
@meshy since it is from https://bitbucket.org/incuna/incuna-txtlocal which is already public I see little point in making your changes private.

File: txtlocal/utils.py [hunk 1]
```diff
@@ -1,156 +1,68 @@
...
+        sender: Must be word up to 11 characters or a number up to 14 digits.
+                Defaults to settings.DEFAULT_FROM_SMS. (Must be a string.)
+        username: Defaults to settings.TXTLOCAL_USERNAME.
+        password: Defaults to settings.TXTLOCAL_PASSWORD.
+        **kwargs: Will be passed through to textlocal in the POST data.
+
+    Any unrecognised kwargs will be passed to txtlocal in the POST data.
     """
+    post_data = {
+        'selectednums': '.'.join(recipient_list),
```
Comment (maxpeterson) [inline txtlocal/utils.py]: Is this correct?

**Comment by ghickman** [comment]
@meshy – The changes you've made look ok at a cursory glance. Could you fix the `setup.py` while you're poking around?

File: txtlocal/utils.py (hunk 1 above)
Comment (meshy) [inline txtlocal/utils.py]: nope :)

**Comment by jturnbull** [comment]
Needs a LICENCE and a CHANGELOG

**Comment by meshy** [comment]
Ok, will also add a README.

**Comment by meshy** [comment]
fwiw... also needs tests :(

File: setup.py [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
+    packages=find_packages(),
     include_package_data=True,
-    install_requires=[
-    ],
-    version = "1.0",
-    description = "Incuna app for sending SMS messages via http://www.txtlocal.co.uk",
-    author = "Incuna Ltd",
-    author_email = "admin@incuna.com",
-    url = "http://incuna.com/",
+    install_requires=['requests'],
```
Comment (ghickman) [inline setup.py]: This should have a `>=` on it.

Comment (meshy) [inline setup.py]: Any suggestions for required version?

Comment (ghickman) [inline setup.py]: Whatever you used to develop it =)

**Comment by ghickman** [comment]
Couple of pedantic tweaks ready for rolling a release.

@maxpeterson – Are you happy to merge this now?

**Comment by maxpeterson** [comment]
@ghickman it looks OK but has it actually been tested - as in sending an SMS?

**Comment by ghickman** [comment]
@meshy – Interesting, that makes no sense to me… as you URI encode key values!

**Pull request merged**